frontend/api/drugSearch.ts

```
@@ -0,0 +1,230 @@
+import useUserLocation from '@/hooks/use-user-location';
+import { setCacheEntry } from '../redux/drugSearchSlice';
+import { store } from '../redux/store';
+import { distanceBetweenCoordinates, zipToCoords } from './distance';
+import { dsntClient } from './dsntClient';
+import { groupID, ndcOverride, scriptSaveClient } from './scriptSaveClient';
+import {
+  type DrugSearchResult,
+  DsntPriceRequest,
+  ScriptSaveFindDrugsResponse,
+  ScriptSavePriceRequest,
+} from './types';
+
+const MAX_DSNT_RADIUS = 125.99;
+const MAX_SCRIPTSAVE_NUMRESULTS = 999999999;
+
+/**
+ * This function queries both adjudicator APIs to find pharmacy drugs with
+ * the given 'drugName', within 'radius' miles of 'zipCode'. It de-duplicates
+ * results by pharmacy, selecting results with the lowest price for a given
+ * pharmacy. The function is used by wrappers to return sorted search results.
+ *
+ * @param drugName of the drug to search.
+ * @param zipCode to search for pharmacy drugs in.
+ * @param radius to search within, relative to 'zipCode'.
+ */
+async function searchDrug(
+  drugName: string,
+  radius: number,
+  includeGeneric: boolean,
+  zipCode?: number
+) {
+  // Get user's location (lat/lon/zip) from hook
+  const { userLat, userLon, userZipCode, userLocationError } = await useUserLocation();
```

You should throw the error your function rather than return it in the results, especially if you're going to throw the error anyways